src/CommunityToolkit.Aspire.Hosting.Dapr.AzureRedis/AzureRedisCacheDaprHostingExtensions.cs

```diff
@@ -0,0 +1,140 @@
+﻿using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Azure;
+using Aspire.Hosting.Dapr;
+using Azure.Provisioning;
+using Azure.Provisioning.AppContainers;
+using Azure.Provisioning.Expressions;
+using Azure.Provisioning.KeyVault;
+using AzureRedisResource = Azure.Provisioning.Redis.RedisResource;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for configuring Dapr components with Azure Redis.
+/// </summary>
+public static class AzureRedisCacheDaprHostingExtensions
+{
+    private const string redisDaprState = nameof(redisDaprState);
+    /// <summary>
+    /// Configures a Dapr component resource to use an Azure Redis cache resource.
+    /// </summary>
+    /// <param name="builder">The Dapr component resource builder.</param>
+    /// <param name="source">The Azure Redis cache resource builder.</param>
+    /// <returns>The updated Dapr component resource builder.</returns>
+    public static IResourceBuilder<IDaprComponentResource> WithReference(
+        this IResourceBuilder<IDaprComponentResource> builder,
+        IResourceBuilder<AzureRedisCacheResource> source) =>
+        builder.ApplicationBuilder.ExecutionContext.IsRunMode ? builder : builder.Resource.Type switch
+        {
+            "state" => builder.ConfigureRedisStateComponent(source),
+            _ => throw new InvalidOperationException($"Unsupported Dapr component type: {builder.Resource.Type}"),
+        };
+
+    /// <summary>
+    /// Configures the Redis state component for the Dapr component resource.
+    /// </summary>
+    /// <param name="builder">The Dapr component resource builder.</param>
+    /// <param name="source">The Azure Redis cache resource builder.</param>
+    /// <returns>The updated Dapr component resource builder.</returns>
+    private static IResourceBuilder<IDaprComponentResource> ConfigureRedisStateComponent(
+        this IResourceBuilder<IDaprComponentResource> builder,
+        IResourceBuilder<AzureRedisCacheResource> source)
+    {
+        var daprComponent = AzureDaprHostingExtensions.CreateDaprComponent(redisDaprState, "state.redis", "v1.0");
+
+        var redisHost = new ProvisioningParameter("redisHost", typeof(string));
+        var principalIdParameter = new ProvisioningParameter(AzureBicepResource.KnownParameters.PrincipalId, typeof(string));
+
+        var configureInfrastructure = AzureDaprHostingExtensions.GetInfrastructureConfigurationAction(daprComponent, [redisHost]);
```

Comment:
Ah looks like this is where the stuff I was question if it should be public or not is being used across assemblies.

I have a larger thought that I might roll up separately.